osu.Android/OsuGameActivity.cs

```diff
@@ -48,41 +51,64 @@ private void handleIntent(Intent intent)
             {
                 case Intent.ActionDefault:
                     if (intent.Scheme == ContentResolver.SchemeContent)
-                        handleImportFromUri(intent.Data);
+                        handleImportFromUris(intent.Data);
                     break;
 
                 case Intent.ActionSend:
                 {
                     var content = intent.ClipData?.GetItemAt(0);
                     if (content != null)
-                        handleImportFromUri(content.Uri);
+                        handleImportFromUris(content.Uri);
+                    break;
+                }
+
+                case Intent.ActionSendMultiple:
+                {
+                    var uris = new List<Uri>();
+                    for (int i = 0; i < intent.ClipData?.ItemCount; i++)
+                    {
+                        var content = intent.ClipData?.GetItemAt(i);
+                        if (content != null)
+                            uris.Add(content.Uri);
+                    }
+                    handleImportFromUris(uris.ToArray());
                     break;
                 }
             }
         }
 
-        private void handleImportFromUri(Uri uri) => Task.Factory.StartNew(async () =>
+        private void handleImportFromUris(params Uri[] uris) => Task.Factory.StartNew(async () =>
         {
-            // there are more performant overloads of this method, but this one is the most backwards-compatible
-            // (dates back to API 1).
-            var cursor = ContentResolver?.Query(uri, null, null, null, null);
+            var tasks = new List<ImportTask>();
 
-            if (cursor == null)
-                return;
+            await Task.WhenAll(uris.Select(async uri =>
```

Comment:
yeah, no wonder this is holding up the game launch. i find this sort of dodgy, you're likely starving the TPL thread pool with that sort of code.

did you try creating these tasks serially via continuations (ie. one by one instead of a free-for-all like this)?

Reply:
Actually I was suspecting that one to be retarding the game launch, but upon inspection, resolving all filenames + opening the file streams and copying them to a memory stream takes about 1.5s for 20 beatmaps. Will still look into what you proposed out of curiosity

Reply:
I still get results lying in a  similar time frame when running them sequentially one by one so I believe the problem lies somewhere else. Upon looking at `ArchiveModelManager` :https://github.com/ppy/osu/blob/4fba0c8e6a3279216294a5a75fa2f79687d0a0c1/osu.Game/Database/ArchiveModelManager.cs#L136 this line is  spawning a task for each import task which may deplete the TPL pool and we're limiting the actual number of imports running to one concurrent import further down, so the problem may lie there.

Reply:
After replacing that exact line to runs tasks sequentially I can affirm the line aforementionned causes the game loading to be longer when importing a lot of beatmaps. Opinions welcome on how to handle that elegantly (as I don't think putting a platform check to chose whether to run tasks sequentially or in parallel in the middle of the code is the best idea)

Reply:
hmm, so the diagnosis was correct, just not the place... a way to change this *might* be to have a dedicated separate pool for imports, although I'm not sure whether spawning even more threads on mobile is a good idea. maybe the tasks should run sequentially during launch and then switch to fully parallelised while the game is ready to go? not sure.

I probably wouldn't base it on any platform check since this will likely behave similarly everywhere, it's just it's been a rarely-checked scenario until now (albeit possible, as the game accepts filenames to import via command line).

Reply:
I don't think either spawning one more thread per archive manager is a good idea, so alternatively I guess the game could maybe use a custom threaded task scheduler for loading menu components asynchronously instead .

Honestly I'm tempted to just make import tasks running sequentially but it is easily foreseeable this will reduce the import performance when running large imports so .. dunno .

Reply:
Something like this?

https://github.com/ppy/osu/blob/3e3d5b2e37431d5e6074fbcb84ca58a7adae0cbe/osu.Game/Database/ArchiveModelManager.cs#L48

Reply:
Yet another alternative would be to schedule all imports tasks on `import_scheduler` but not sure how much it would tank performance again ...

Reply:
This issue also seems out of scope. I'd open a separate issue covering this (since it looks like it would probably fail on desktop import as well?) and continue with getting this PR merged as-is.

Reply:
This is now tracked by #11251. As such, marking this as resolved.